app/unity/Assets/Scripts/FadeScript.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A component to be attached to the game object that is able to gradually change the alpha parameter 
+/// from the CanvasGroup component of the same object.
+/// </summary>
+public class FadeScript : MonoBehaviour
+{
+    /// <summary>
+    /// The speed of alpha transition. The higher the number - the faster the transition.
+    /// </summary>
+    public float fadeSpeed = 1;
+    /// <summary>
+    /// If true - fade in. If false - fade out.
+    /// </summary>
+    public bool fadeDirection = true;
+    /// <summary>
+    /// A CanvasGroup component attached to the same game object as this script.
+    /// The alpha property of this component is being manipulated by this script.
+    /// </summary>
+    private CanvasGroup canvasGroup;
+
+    /// <summary>
+    /// Called when the script object is initialized.
+    /// </summary>
+    private void Awake()
+    {
+        canvasGroup = GetComponent<CanvasGroup>();
+    }
+
+    /// <summary>
+    /// Switches the fade direction to true - fade in.
+    /// The Update calls handle the actual transition.
+    /// </summary>
+    public void FadeIn()
+    {
+        fadeDirection = true;
```

Comment:
With the Canvas Group, please add enabling Interactable and Blocks Raycasts.

See: https://docs.unity3d.com/Packages/com.unity.ugui@1.0/manual/class-CanvasGroup.html#:~:text=The%20Canvas%20Group%20can%20be,as%20well%20as%20all%20children.

Reply:
Belay that, as  we talked the GameState machine should handle enabling and disabling the canvas group.